src/api/exchange/DepositApiImpl.ts

```diff
@@ -0,0 +1,186 @@
+    return getEpochAddressForNetwork(networkId)
+  }
+
+  public async getBatchTime(): Promise<number> {
+    const contract = await this._getContract()
+    const BATCH_TIME = await contract.methods.BATCH_TIME().call()
+    return +BATCH_TIME
+  }
+
+  public async getCurrentBatchId(): Promise<number> {
+    const contract = await this._getContract()
+    const batchId = await contract.methods.getCurrentBatchId().call()
+    return +batchId
+  }
+
+  public async getSecondsRemainingInBatch(): Promise<number> {
+    const contract = await this._getContract()
+    const secondsRemainingInBatch = await contract.methods.getSecondsRemainingInBatch().call()
+    return +secondsRemainingInBatch
+  }
+
+  public async getBalance(userAddress: string, tokenAddress: string): Promise<BN> {
+    if (!userAddress || !tokenAddress) return ZERO
+
+    const contract = await this._getContract()
+    const balance = await contract.methods.getBalance(userAddress, tokenAddress).call()
+
+    return toBN(balance)
+  }
+
+  public async getPendingDepositAmount(userAddress: string, tokenAddress: string): Promise<BN> {
+    if (!userAddress || !tokenAddress) return ZERO
+
+    const contract = await this._getContract()
+    const depositAmount = await contract.methods.getPendingDepositAmount(userAddress, tokenAddress).call()
+
+    return toBN(depositAmount)
+  }
+
+  public async getPendingDepositBatchId(userAddress: string, tokenAddress: string): Promise<number> {
+    if (!userAddress || !tokenAddress) return 0
+
+    const contract = await this._getContract()
+    const depositBatchId = await contract.methods.getPendingDepositBatchNumber(userAddress, tokenAddress).call()
+
+    return +depositBatchId
+  }
+
+  public async getPendingWithdrawAmount(userAddress: string, tokenAddress: string): Promise<BN> {
+    if (!userAddress || !tokenAddress) return ZERO
+
+    const contract = await this._getContract()
+    const withdrawAmount = await contract.methods.getPendingWithdrawAmount(userAddress, tokenAddress).call()
+
+    return toBN(withdrawAmount)
+  }
+
+  public async getPendingWithdrawBatchId(userAddress: string, tokenAddress: string): Promise<number> {
+    if (!userAddress || !tokenAddress) return 0
+
+    const contract = await this._getContract()
+    const withdrawBatchId = await contract.methods.getPendingWithdrawBatchNumber(userAddress, tokenAddress).call()
+
+    return +withdrawBatchId
+  }
+
+  public async deposit(
+    userAddress: string,
+    tokenAddress: string,
+    amount: BN,
+    txOptionalParams?: TxOptionalParams,
+  ): Promise<Receipt> {
+    const contract = await this._getContract()
+    const tx = contract.methods.deposit(tokenAddress, amount).send({ from: userAddress })
+
+    if (txOptionalParams && txOptionalParams.onSentTransaction) {
+      tx.once('receipt', txOptionalParams.onSentTransaction)
+    }
+
+    log(`[DepositApiMock] Deposited ${formatAmount(amount)} for token ${tokenAddress}. User ${userAddress}`)
```

Comment:
Not `Mock`. 
And `formatAmount` will only work for tokens with decimals == 18.
I recommend to either use the token decimals (I guess not available here) or just log the raw amount.

Reply:
:+1: Done